src/app/loadout-drawer/loadout-item-conversion.ts

```diff
@@ -44,15 +45,44 @@ export function getItemsFromLoadoutItems(
       const modsForBucket =
         loadoutItem.equip && modsByBucket ? modsByBucket[item.bucket.hash] ?? [] : [];
 
-      let overrides = loadoutItem.socketOverrides;
+      let overrides = loadoutItem.socketOverrides || {};
 
+      // Apply armour mods to the socket overrides
       for (const modHash of modsForBucket) {
         const socket = item.sockets?.allSockets.find((s) => plugFitsIntoSocket(s, modHash));
         if (socket) {
           overrides = { ...overrides, [socket?.socketIndex]: modHash };
         }
       }
 
+      // Hydrate the empty sockets for fragments as overrides. This ensures that the stats that
+      // result from the overridden item are correct.
+      if (
+        item.bucket.hash === BucketHashes.Subclass &&
+        item.sockets?.allSockets &&
+        defs.isDestiny2()
+      ) {
+        const aspects = getSocketsByCategoryHash(item.sockets, SocketCategoryHashes.Aspects);
+        let activeFragments = 0;
+        for (const aspect of aspects) {
+          activeFragments += aspect.plugged?.plugDef.plug.energyCapacity?.capacityValue || 0;
+        }
+
+        const fragments = getSocketsByCategoryHash(item.sockets, SocketCategoryHashes.Fragments);
+        for (const fragment of fragments) {
+          if (overrides[fragment.socketIndex] && activeFragments) {
+            activeFragments--;
+            continue;
+          }
+
+          fragment.actuallyPlugged = fragment.plugged || undefined;
```

Comment:
I am thinking we could maybe use this to indicate the "inactive" fragments in the UI. It's a bit of a hack but we can essentially check to see if the override is the empty plug, and if so show the inactive `actuallyPlugged`. Thoughts?